CenterForOpenScience/osf.io
# [ENG-2107] Clean up withdrawn preprint and registration files

## Purpose

This PR is meant to address the fate of files of withdrawn registration, it also fixes a few TODOs that were left in the code base surrounding withdraws. 

## Changes

- make periodic task to make withdrawn files as deleted
- make periodic task to ensure deleted files are purged after 30 days
- add tests with mock google cloud drive, changes dome settings to that effect.=


TODOS:
  - clean-up/optimize `get_descendants_recursive`
  - Remove unused parent param on BaseFIleNode .delete





## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify that withdrawn files are marked as deleted
- Verify deleted withdrawn files are purged

## Documentation



New behavior, no docs.

## Side Effects

This purges files and effects Product sensitive behavior. 

## Ticket

https://openscience.atlassian.net/browse/https://openscience.atlassian.net/browse/ENG-2107
